scripts/scr_unit_size/scr_unit_size.gml

```diff
@@ -1,23 +1,30 @@
-
-function scr_unit_size(armour, role, other_factors) {
+
+// lets face it this should all be solved in a method in the marines struct....
+function scr_unit_size(armour, role, other_factors, mobility) {
 
     // armour: armour
     // role: role
     // other_factors: +?
+	//mobility item
     var sz = 1;
     if (role!=""){
-
-        if (string_count("Dread",armour)>0) {sz=5;}
-        else if (role="Rhino") {sz=10;}
+		var bulky_armour = ["Terminator Armour", "Tartaros"]
+        if (string_count("Dread",armour)>0) {
+		sz+=5;
+	}else if (array_contains(bulky_armour,armour)){sz +=1};
+        if (role="Rhino") {sz=10;}
         else if (role="Predator") {sz=10;}
         else if (role="Land Raider") {sz=20;} 
         else if (role="Land Speeder") {sz=5;}
         else if (role="Whirlwind") {sz=10;}
         else if (role="Harlequin Troupe") {sz=5;}
+		else if (role="Chapter Master"){sz+=1;}

```

Comment:
you can use `sz++` instead of `sz+=1`, but it's a style issue basically, your choice

Reply:
honestly think i'm gonna redo all of this file as it's pretty much all copied from what was there, and i'm not happy with it at all.

Reply:
sure, if you're not satisfied with the code, then by all means do a redo